### ISSUE TITLE
Profile and remove dead optimizations

### ISSUE THREAD

**Issue by benjamingr** [primus/eventemitter3#109]
Hey, a lot of the performance optimizations here assume things about V8 that are no longer true with TurboFan.

This issue is to track and remove those eventually.

**Comment by lpinca** [comment]
Yes, we need to revisit some micro optimizations once TF+I lands (8.2.0?).
For example we can use `Object.create(null)` directly now instead of

```js
function Storage() {}
Storage.prototype = Object.create(null);
```

which is nice as it will remove some feature detection code reducing the library size.

If you want to help tracing and removing dead optimizations please do, it will be greatly appreciated :)
